src/Compilers/CSharp/Portable/Symbols/Synthesized/Records/SynthesizedRecordPrintMembers.cs

```diff
@@ -0,0 +1,277 @@
+                            ordinal: 0, RefKind.None, "builder", isDiscard: false, Locations)),
+                    IsVararg: false,
+                    DeclaredConstraintsForOverrideOrImplementation: ImmutableArray<TypeParameterConstraintClause>.Empty);
+        }
+
+        protected override int GetParameterCountFromSyntax() => 1;
+
+        internal override void GenerateMethodBody(TypeCompilationState compilationState, DiagnosticBag diagnostics)
+        {
+            var F = new SyntheticBoundNodeFactory(this, ContainingType.GetNonNullSyntaxNode(), compilationState, diagnostics);
+            try
+            {
+                ImmutableArray<Symbol> printableMembers = ContainingType.GetMembers()
+                    .WhereAsArray(m => m.DeclaredAccessibility == Accessibility.Public && (m.Kind == SymbolKind.Field || m.Kind == SymbolKind.Property));
+
+                if (printableMembers.Any((m, _) => SymbolExtensions.GetTypeOrReturnType(m).Type.IsErrorType(), arg: (object?)null))
+                {
+                    F.CloseMethod(F.ThrowNull());
+                    return;
+                }
+
+                if (ReturnType.IsErrorType())
+                {
+                    F.CloseMethod(F.ThrowNull());
+                    return;
+                }
+
+                ArrayBuilder<BoundStatement>? block = printableMembers.IsEmpty ? null : ArrayBuilder<BoundStatement>.GetInstance();
+                BoundParameter builder = F.Parameter(this.Parameters[0]);
+                if (ContainingType.BaseTypeNoUseSiteDiagnostics.IsObjectType())
+                {
+                    if (printableMembers.IsEmpty)
+                    {
+                        // return false;
+                        F.CloseMethod(F.Return(F.Literal(false)));
+                        return;
+                    }
+                }
+                else
+                {
+                    MethodSymbol? printMethod = FindValidPrintMethod(ContainingType.BaseTypeNoUseSiteDiagnostics, DeclaringCompilation);
+                    if (printMethod is null)
+                    {
+                        F.CloseMethod(F.ThrowNull()); // an error was reported in base checks already
+                        return;
+                    }
+
+                    var basePrintCall = F.Call(receiver: F.Base(ContainingType.BaseTypeNoUseSiteDiagnostics), printMethod, builder);
+                    if (printableMembers.IsEmpty)
+                    {
+                        // return base.print(builder);
+                        F.CloseMethod(F.Return(basePrintCall));
+                        return;
+                    }
+                    else
+                    {
+                        // if (base.print(builder))
+                        //     builder.Append(", ")
+                        block!.Add(
```

Comment:
consider using `Debug.Assert` instead.

Reply:
The reason I used `!` here is because we have an explicit check above in the method which ensures we don't have null `block` here.

---
In reply to: [469407902](https://github.com/dotnet/roslyn/pull/46616#discussion_r469407902) [](ancestors = 469407902)